From 3e00b1bab000cbd054b499414774d9a9527e2f3a Mon Sep 17 00:00:00 2001
From: "R, Kannappan" <r.kannappan@intel.com>
Date: Thu, 25 Jun 2015 19:58:52 +0530
Subject: [PATCH] pi3usb-mux: put the chip to low power mode during suspend

while entering suspend, put the chip to low power mode and
during resume, configure the mux to the original state before
suspend

Change-Id: I0daaf94910025b96ece283538b7c02e953222297
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-40217
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
---
 drivers/misc/pi3usb30532_mux.c | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/pi3usb30532_mux.c b/drivers/misc/pi3usb30532_mux.c
index 6bfc6ab97b9e..951f6a7e17a5 100644
--- a/drivers/misc/pi3usb30532_mux.c
+++ b/drivers/misc/pi3usb30532_mux.c
@@ -55,6 +55,7 @@ struct pi3usb30532_mux {
 	struct work_struct mux_work;
 	struct mutex event_lock;
 	struct typec_phy *phy;
+	u8 cur_config;
 };
 
 /* read/write/modify pi3usb30532 register values */
@@ -140,10 +141,12 @@ static void pi3usb30532_mux_event_worker(struct work_struct *work)
 		break;
 	case TYPEC_POS_DISCONNECT:
 	default:
-		conf = PI3USBMUX_OPEN_ONLYNOPD;
+		conf = PI3USBMUX_OPEN_WITHPD;
 		break;
 	}
 
+	chip->cur_config = conf;
+
 	ret = pi3usb30532_mux_sel_ctrl(chip, conf);
 	mutex_unlock(&chip->event_lock);
 }
@@ -245,14 +248,25 @@ static int pi3usb30532_suspend(struct device *dev)
 {
 	struct pi3usb30532_mux *chip = dev_get_drvdata(dev);
 
+	int ret;
+
+	mutex_lock(&chip->event_lock);
+	ret = pi3usb30532_mux_write_reg(chip->client,
+				PI3USB30532_SEL_CTRL_REG,
+				PI3USBMUX_OPEN_WITHPD);
+	mutex_unlock(&chip->event_lock);
 	dev_dbg(&chip->client->dev, "pi3usb30532 suspend\n");
-	return 0;
+	return ret;
 }
 
 static int pi3usb30532_resume(struct device *dev)
 {
 	struct pi3usb30532_mux *chip = dev_get_drvdata(dev);
 
+	mutex_lock(&chip->event_lock);
+	pi3usb30532_mux_sel_ctrl(chip, chip->cur_config);
+	mutex_unlock(&chip->event_lock);
+
 	dev_dbg(&chip->client->dev, "pi3usb30532 resume\n");
 	return 0;
 }
-- 
1.9.1

